OpenRA.Game/WorldUtils.cs

```diff
@@ -41,6 +41,113 @@ public static IEnumerable<Actor> FindActorsInCircle(this World world, WPos origi
+		/// TODO: Update this to take the z offsets into account, allowing for 3D raycasting.
+		/// </summary>
+		/// <param name="world">The engine world the raycast is to be done in.</param>
+		/// <param name="target">The position the ray should end at</param>
+		/// <param name="source">The position the ray should start at</param>
+		/// <param name="rayWidth">How close an actor's health radius needs to be to the ray to be considered 'intersected' by the ray</param>
+		/// <returns>A list of all the actors 'intersected' by the ray</returns>
+		public static IEnumerable<Actor> FindActorsInRaycast(this World world, WPos target, WPos source, WDist rayWidth)
+		{
+			// This raycast is done by first just finding all actors within a square that starts at the source, and ends at the target.
+			// Then we iterate over this list, and find all actors for which their health radius is at least within rayWidth of the ray.
+			// For actors without a health radius, we simply check their center point.
+			// The square in which we select all actors must be large enough to encompass the entire ray's width.
+			var xOver = 0;
+			var yOver = 0;
+			if (target.X < source.X)
+				xOver = -1024;
+			else
+				xOver = 1024;
+
+			if (target.Y < source.Y)
+				yOver = -1024;
+			else
+				yOver = 1024;
+
+			WVec dir = new WVec(xOver, yOver, 0);
+			var overselect = dir * (1 + (rayWidth.Length / 1024));
+			var finalTarget = target + overselect;
+			var finalSource = source - overselect;
+
+			var actorsInSquare = world.ActorMap.ActorsInBox(finalTarget, finalSource);
+			var intersectedActors = new List<Actor>();
+
+			foreach (var currActor in actorsInSquare)
+			{
+				long actorWidth;
+				var healthInfo = currActor.Info.TraitInfoOrDefault<HealthInfo>();
+				if (healthInfo != null)
+					actorWidth = healthInfo.Radius.Length;
+				else
+					actorWidth = 0;
+
+				var distance = MinimumDistancePointAndVector(target, source, currActor.CenterPosition);
+				var maxReach = (long)(actorWidth + rayWidth.Length);
+				maxReach *= maxReach;
+
+				// Log.Write("debug", "FindActorsInRaycast, distance : {0}, maxReach : {1}", distance, maxReach);
+				if (distance <= maxReach)
+					intersectedActors.Add(currActor);
+			}
+
+			return intersectedActors;
+		}
+
+		/// <summary>
+		/// Returns the (squared) minimum distance between a line and a point.
+		/// From http://stackoverflow.com/questions/849211/shortest-distance-between-a-point-and-a-line-segment
+		/// </summary>
+		/// <param name="target">The target point (head) of the line</param>
+		/// <param name="source">The source point (tail) of the line</param>
+		/// <param name="point">The point that the minimum distance should be found to</param>
+		/// <returns>The square of the minimum distance between the line and point.</returns>
+		private static long MinimumDistancePointAndVector(WPos target, WPos source, WPos point)
```

Comment:
This method returns a squared distance, so the name needs to reflect this.  I have edited my suggestion above to match.